Modules/Server/Sources/PocketCastsServer/Public/Search/PodcastSearchTask.swift

```diff
@@ -0,0 +1,29 @@
+import Foundation
+
+public struct PodcastSearchResult: Decodable {
+    public let uuid: String
+    public let title: String
+    public let author: String
+}
+
+public class PodcastSearchTask {
+    var session = URLSession.shared
+
+    public init() {}
+
+    public func search(term: String) async throws -> [PodcastSearchResult] {
+        let searchURL = URL(string: "\(ServerConstants.Urls.cache())discover/search")!
+        var request = URLRequest(url: searchURL)
+        request.httpMethod = "POST"
+
+        let json: [String: Any] = ["term": term]
+
+        let jsonData = try? JSONSerialization.data(withJSONObject: json)
```

Comment:
Same as from the episode task:

```suggestion
        let jsonData = try JSONSerialization.data(withJSONObject: json)
```